### ISSUE TITLE
Add CPU Feature detection to Android NDK for v8

### ISSUE THREAD

**Issue by paul99** [v8mips/v8mips#46]
Please take a look at the below 3 URLs. The first presumed there was a current inability to read /proc/cpuinfo, to get ARM cpu feature info for v8. It turns out that it is OK now, but they think it will be restricted in the future. 

This is likely to be important for MIPS on Chromium, as even with the deprecating of the soft-float mode, since we still do attempt to parse /proc/cpuinfo for FPU presence.

I'm open to a simpler solution, if there is one, otherwise we should proactively add the needed features to the NDK, as they did for arm.

http://code.google.com/p/v8/issues/detail?id=2597
https://code.google.com/p/android/issues/detail?id=53689
https://android-review.googlesource.com/#/c/54900/

**Comment by dusmil** [comment]
Generally, "digit" guy from chromium team already proposes solutions for v8 and NDK, that we should follow if they are landed, but not with urgent priority.
I do not think this CPU features check in android and implementing them is necessarily required for current state of v8 for MIPS. We only do runtime check for FPU feature that becomes useless after soft-float removal commit. Flags passed by compilers supersede the runtime check, so we can easily remove runtime check if proposed patch for V8 is landed for Arm. I do not say that implementing feature check in NDK for mips is useless, it is good to have something similar as it is done for ARM, but it is not currently required by mips port of v8. Maybe in future, ndk CPU feature check will be used in v8 for mips, but now I cannot see what MIPS CPU features we would check in v8.

Bottom line, ndk cpu feature detection is good to have for future, but not urgently required to be implemented nor necessarily required by current v8 mips port.

**Comment by paul99** [comment]
I agree that this is not required today. V8 for arm only works today if it can parse /proc/cpuinfo, same as us.

It will be very quick to get the necessary port into V8 for mips, when the arm side changes.

However, I had thought that it might not be quite so quick to get the android side ported, especially if it happens when we are nearing a release (like now with 4.3, or later with the K-release). And also I guess, NDK releases are on their own slow timeline. So when we need it, we may not be able to add it.

I thought it might be a good idea to get the basic reading of cpu features for mips into Android, like Digit's NDK patch, just so that it is available when the transition to sandboxing arrives. 

We did debate whether we should even keep the check for the Cpu Feature FPU bit, and you did convince me that was a good idea. So maybe we should add that to NDK, just as a precaution.
